cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -115,6 +117,11 @@ controller, and serviceaccounts controller.`,
 		Run: func(cmd *cobra.Command, args []string) {
 			verflag.PrintAndExitIfRequested()
 
+			if err := SetUpPreferredHostForOpenShift(s); err != nil {
```

Comment:
Can this be part of below `ShimForOpenShift`?

Reply:
no, because we need to set a custom RoundTripper before calling `s.Config`